From c7e00dd5d18665bb4c56789f8ed951cbec3bb84f Mon Sep 17 00:00:00 2001
From: Sanyog Kale <skale@habana.ai>
Date: Mon, 24 Jan 2022 17:50:38 +0200
Subject: [PATCH] [SW-58071] gaudi2: use pre compiled rom symbols

Since the addition of gaudi3 changes in ROM code base, the code
compiled for gaudi2 ROM doesn't match exactly with ROM code on
device due to which ppboot fails to find exact symbols of ROM
when ran on device. In order to fix this issue, this patch add
change to use pre-compiled ROM symbols for ppboot compilation which
exactly matches with ROM symbols on device.

Change-Id: I50dcb91181fa2a4dc86f4388a54a8d4bb8ca7b39
---

diff --git a/arch/gaudi2/arc/Makefile b/arch/gaudi2/arc/Makefile
index fe5f611..cf98f06 100644
--- a/arch/gaudi2/arc/Makefile
+++ b/arch/gaudi2/arc/Makefile
@@ -4,7 +4,7 @@
 PPBOOT_MAX_SIZE := 16384
 PREBOOT_MAX_SIZE := 64512
 
-LDFLAGS += -Wl,--just-symbols $(PREBOOT_ROOT_DIR)/../arcboot-custom/exports.elf
+LDFLAGS += -Wl,--just-symbols $(HL_DEV_ARCH)/exports.elf
 
 PREBOOT_SRCS_C += $(shell ls $(PREBOOT_ROOT_DIR)/lib/preboot/thermal_v2.c)
 
diff --git a/arch/gaudi2/arc/exports.elf b/arch/gaudi2/arc/exports.elf
new file mode 100644
index 0000000..cd3abf1
--- /dev/null
+++ b/arch/gaudi2/arc/exports.elf
Binary files differ
